subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -495,6 +500,22 @@ struct ztest_unit_test *ztest_get_next_test(const char *suite, struct ztest_unit
 	return NULL;
 }
 
+bool z_ztest_should_test_run(const char *suite, const char *test)
```

Comment:
If we're not running in POSIX this is an extra function call that should always return `true`. Can we split this implementation off? I can see 2 alternatives (I personally like the second but I'm open to either or a 3rd option).

1. Use `#ifdef CONFIG_ARCH_POSIX` to either have an implementation or a static inline version that just returns `true`
2. Put the implementation in a separate `.c` file and conditionally include it in the CMakeLists.txt based on the same config. I like this option better because I'm not a fan of having large `#if/else` statements that conditionally include functions, I much prefer to split those off into compilational units.